lib/util/exec/hermit.ts

```diff
@@ -0,0 +1,51 @@
+import os from 'os';
+import upath from 'upath';
+import { GlobalConfig } from '../../config/global';
+import { logger } from '../../logger';
+import { findUpLocal } from '../fs';
+import { rawExec } from './common';
+import type { RawExecOptions } from './types';
+
+export function isHermit(): boolean {
+  const { binarySource } = GlobalConfig.get();
+  return binarySource === 'hermit';
+}
+
+export async function findHermitCwd(cwd: string): Promise<string> {
+  const hermitFile = await findUpLocal('bin/hermit', upath.join(cwd));
+
+  if (hermitFile === null) {
+    throw new Error(`hermit not found for ${cwd}`);
+  }
+
+  return upath.join(GlobalConfig.get('localDir'), upath.dirname(hermitFile));
+}
+
+export async function getHermitEnvs(
+  rawOptions: RawExecOptions
+): Promise<Record<string, string>> {
+  const cwd = rawOptions.cwd?.toString() ?? '';
+  const hermitCwd = await findHermitCwd(cwd);
+  logger.debug({ cwd, hermitCwd }, 'fetching hermit environment variables');
+  // with -r will output the raw unquoted environment variables to consume
+  const hermitEnvResp = await rawExec('./hermit env -r', {
```

Comment:
how does hermit ensure that the binaries are available and callable?

does it use wrapper scripts to unpack the files from repo on the fly?

Reply:
Yes. All calls are eventually forward into the Hermit application, which does the installation/unpack on the fly.